docs/framework/architecture/ui-library.md

```diff
@@ -439,6 +441,355 @@ dropdownView.bind( 'isEnabled' ).toMany( buttons, 'isEnabled',
+
+<info-box warning>
+	Below you'll learn about the ways to disable some of these methods, but remember to always leave at least one option to close the modal. Otherwise the users will be locked inside.
+</info-box>
+
+#### Structure and behaviour
+
+A dialog can have three parts, each of which is optional: the [header](#header) (also used as a drag handler), the [content](#content) (the body of the dialog) and the [action buttons](#action-buttons) area (a collection of buttons). The order of the parts can not be changed.
+
+##### Header
+
+A header may consist of any combination of three elements:
+
+* the icon,
+* the title,
+* the "Close" button.
+
+By default, the "Close" button ("X") is added to the header as long as you provide an icon or a title. To hide it, set `hasCloseButton` flag to `false`:
+
+```js
+import { icons } from 'ckeditor5/src/core.js';
+
+// ...
+
+editor.plugins.get( 'Dialog' ).show( {
+	icon: icons.pencil,
+	title: 'My first dialog',
+	// Do not display the "Close" button.
+	hasCloseButton: false,
+
+	// The rest of the dialog definition...
+} );
+```
+
+<info-box>
+	If you decide to hide the "Close" button, remember to leave some other way(s) to close the dialog (like pressing the <kbd>Esc</kbd> key).
+</info-box>
+
+##### Content
+
+This part can be a single [view](#views) or a collection of views. They will be displayed directly inside a body of a dialog. You can find an example of how to insert a block of text into the dialog.
+
+```js
+const textView = new View( locale );
+
+textView.setTemplate( {
+	tag: 'div',
+	attributes: {
+		style: {
+			padding: 'var(--ck-spacing-large)',
+			whiteSpace: 'initial',
+			width: '100%',
+			maxWidth: '500px'
+		},
+		tabindex: -1
+	},
+	children: [
+		'This is a sample content of the dialog.',
+		'You can put here text, images, inputs, buttons, etc.'
+	]
+} );
+
+editor.plugins.get( 'Dialog' ).show( {
+	title: 'Info dialog with text',
+	content: textView,
+
+	// The rest of the dialog definition...
+} );
+```
+
+##### Action buttons
+
+The last piece of the dialog is the actions area, where the buttons are displayed. Their behaviour can be fully customized using the `onCreate()` and `onExecute()` callbacks. Below you can find an example of the configuration for the four custom buttons:
+
+* The "OK" button that closes the dialog and has a custom CSS class set,
+* The "Set custom title" button that changes the dialog's title,
+* The "Drag detector" button that changes its state once the dialog was dragged by the user.
+* The "Cancel" button that closes the dialog.
+
+```js
+editor.plugins.get( 'Dialog' ).show( {
+	// ...
+
+	actionButtons: [
+		{
+			label: 'OK',
+			class: 'ck-button-action',
+			withText: true,
+			onExecute: () => dialog.hide()
+		},
+		{
+			label: 'Set custom title',
+			withText: true,
+			onExecute: () => {
+				dialog.view!.headerView!.label = 'New title';
+			}
+		},
+		{
+			label: 'This button will be disabled in 3...'
```

Comment:
I'd go the other way -- make it enabled in 3, 2, 1. It used to be a pattern in dialogs some time ago (so you won't click by mistake). And when enabled, change label to something like "Enabled after 3 seconds" or simply "Button".